components/automate-ui/src/app/pages/team-add-users/team-add-users.component.ts

```diff
@@ -41,86 +41,79 @@ export class TeamAddUsersComponent implements OnInit, OnDestroy {
   public team: Team;
   public users: User[] = [];
   private mapOfUsersToFilter: HashMapOfUsers = {};
-  public isIAMv2: boolean;
+  public isIAMv2$: Observable<boolean>;
   private isDestroyed = new Subject<boolean>();
   public addingUsers = false;
   private usersToAdd: { [id: string]: User } = {};
   public showSecondary = false;
   public addUsersFailed = '';
-  public loading = true;
+  public loading$: Observable<boolean>;
+  public teamId = '';
 
   constructor(
     private store: Store<NgrxStateAtom>,
     private router: Router) {
-      store.select(isIAMv2).subscribe(latest => this.isIAMv2 = latest);
   }
 
   ngOnInit(): void {
-    this.store.select(isIAMv2)
-      .pipe(takeUntil(this.isDestroyed))
-      .subscribe((version) => {
-        if (version === null) { return; }
-
-        if (this.isIAMv2) {
-          this.store.select(v2TeamFromRoute)
-            .pipe(filter(identity), takeUntil(this.isDestroyed))
-            .subscribe(this.getUsersForTeam.bind(this));
-        } else {
-          this.store.select(v1TeamFromRoute)
-            .pipe(filter(identity), takeUntil(this.isDestroyed))
-            .subscribe(this.getUsersForTeam.bind(this));
-        }
-      });
-
-    this.store.select(routeParams).pipe(
-      pluck('id'),
-      filter(identity),
-      takeUntil(this.isDestroyed))
-      .subscribe((id: string) => {
-        this.store.select(routeURL).pipe(
-          filter(identity),
-          takeUntil(this.isDestroyed))
-          .subscribe((url: string) => {
-            // Only fetch if we are on the team details route, otherwise
-            // we'll trigger GetTeam with the wrong input on any route
-            // away to a page that also uses the :id param.
-            if (TEAM_ADD_USERS_ROUTE.test(url)) {
-              this.store.dispatch(new GetTeam({ id }));
-            }
-          });
+    this.store.dispatch(new GetUsers());
+    this.isIAMv2$ = this.store.select(isIAMv2);
+
+    combineLatest([
+      this.isIAMv2$,
+      this.store.select(v2TeamFromRoute),
+      this.store.select(v1TeamFromRoute)
+    ]).pipe(
+        takeUntil(this.isDestroyed),
+        filter(([isV2, _v2TeamFromRoute, _v1TeamFromRoute]) => isV2 !== null),
+        map(([isV2, v2Team, v1Team]) => isV2 ? [v2Team, v2Team.id] : [v1Team, v1Team.guid]),
+        distinctUntilChanged(
+          ([_teamA, teamIdA]: [Team, string], [_teamB, teamIdB]: [Team, string]) =>
+          teamIdA === teamIdB)
+      ).subscribe(([team, teamId]) => {
+        this.team = team;
+        this.teamId = teamId;
+        this.store.dispatch(new GetTeamUsers({ id: teamId}));
```

Comment:
Found a bug with IAMv1 only: This code is never reached, so `GetTeamUsers` is never dispatched and `this.teamId` remains an empty string. In the browser that manifests via this sequence:
Settings > Teams > [team] > Add Users > [close], then you get the page stuck with the spinner open and the errors shown. The error is reporting on the fact that the team name is `auth:teams:<empty string>` which is illegal syntax.
![image](https://user-images.githubusercontent.com/6817500/68450472-1364f500-01a0-11ea-913c-4d105c3f39fe.png)

Reply:
I can not reproduce this problem.

Reply:
With your later commit, this bug is no longer present. 🎉